systemtest/src/test/java/io/strimzi/systemtest/StrimziUpgradeST.java

```diff
@@ -0,0 +1,186 @@
+            //kubeClient.apply(new File("../install/cluster-operator"));
+            LOGGER.info("Waiting for CO redeployment");
+            kubeClient.waitForDeployment("strimzi-cluster-operator", 1);
+
+            LOGGER.info("Waiting for ZK SS roll");
+            StUtils.waitTillSsHasRolled(client, NAMESPACE, zkSsName, zkPods);
+            LOGGER.info("Checking ZK pods using new image");
+            waitTillAllPodsUseImage(client.apps().statefulSets().inNamespace(NAMESPACE).withName(zkSsName).get().getSpec().getSelector().getMatchLabels(),
+                    "strimzi/zookeeper:latest");
+            LOGGER.info("Waiting for Kafka SS roll");
+            StUtils.waitTillSsHasRolled(client, NAMESPACE, kafkaSsName, kafkaPods);
+            LOGGER.info("Checking Kafka pods using new image");
+            waitTillAllPodsUseImage(client.apps().statefulSets().inNamespace(NAMESPACE).withName(kafkaSsName).get().getSpec().getSelector().getMatchLabels(),
+                    "strimzi/kafka:latest");
+            LOGGER.info("Waiting for EO Dep roll");
+
+            // Check the TO and UO also got upgraded
+            StUtils.waitTillDepHasRolled(client, NAMESPACE, eoDepName, eoPods);
+            LOGGER.info("Checking EO pod using new image");
+            waitTillAllContainersUseImage(
+                    client.extensions().deployments().inNamespace(NAMESPACE).withName(eoDepName).get().getSpec().getSelector().getMatchLabels(),
+                    0,
+                    "strimzi/topic-operator:latest");
+            waitTillAllContainersUseImage(
+                    client.extensions().deployments().inNamespace(NAMESPACE).withName(eoDepName).get().getSpec().getSelector().getMatchLabels(),
+                    1,
+                    "strimzi/user-operator:latest");
+
+
+            // Tidy up
+        } catch (KubeClusterException e) {
+            if (kafkaEphemeralYaml != null) {
+                kubeClient.delete(kafkaEphemeralYaml);
+            }
+            if (coDir != null) {
+                kubeClient.delete(coDir);
+            }
+            throw e;
+        }
+    }
+
+    private void waitTillAllPodsUseImage(Map<String, String> matchLabels, String image) {
+        waitTillAllContainersUseImage(matchLabels, 0, image);
+    }
+
+    private void waitTillAllContainersUseImage(Map<String, String> matchLabels, int container, String image) {
+        TestUtils.waitFor("All pods matching " + matchLabels + " to have image " + image, 1_000, 300_000, () -> {
+            List<Pod> pods1 = client.pods().inNamespace(NAMESPACE).withLabels(matchLabels).list().getItems();
+            for (Pod pod : pods1) {
+                if (!image.equals(pod.getSpec().getContainers().get(container).getImage())) {
+                    return false;
+                }
+            }
+            return true;
+        });
+    }
+
+    private File downloadAndUnzip(String url) throws IOException {
```

Comment:
We already have one system test which downloads zip and then unpacks it. Maybe it will be a good idea to move this logic to the Utils? Link to ST with downloading/unzipping - https://github.com/strimzi/strimzi-kafka-operator/blob/6565df19caec895b4ea4a59fb56dd450127ac032/systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java#L48